### ISSUE TITLE
Keep displayName of the superclass

### ISSUE THREAD

**Issue by fsdiogo** [moxystudio/js-class-is#2]
When logging an instance of the wrapped class, we want to keep the displayName of the superclass.

**Comment by fsdiogo** [comment]
Partially fixed by https://github.com/moxystudio/js-class-is/pull/3.

**Comment by PedroMiguelSS** [comment]
This is actually working good on node. But it should be fixed in order to be working correctly on browsers.

**EDIT** [07/04/2017]
@satazor I've been working on this issue and I found a possible solution. However, it uses _eval()_ function which is not a best practice. Here is how the problem was solved:

```js
function withIs(Class, { className, symbolName }) {
    const symbol = Symbol.for(symbolName);

	eval(`var newClass = class ${className} extends ${Class} {
            constructor(...args) {
                super(...args);
                Object.defineProperty(this, symbol, { value: true });
            }
        }`)

    Object.defineProperty(newClass.prototype, Symbol.toStringTag, {
        value: className,
    });

    const ClassIsWrapper = {
        // The code below assigns the class wrapper to an object to trick
        // JavaScript engines to show the name of the extended class when
        // logging an instances.
        // We are assigning an anonymous class (class wrapper) to the object
        // with key `className` to keep the correct name.
        // If this is not supported it falls back to logging `ClassIsWrapper`.
        [className]: newClass,
    }[className];

    return ClassIsWrapper;
}
```

I also made some performance tests and I think that there is no considerable time execution differences between a declared class with _eval()_ and without _eval()_. 

I did the following tests on chrome console: 
```js
class RectangleX {
	constructor(height, width) {
		this.height = height;
		this.width = width;
    }
	get area() {
		return this.calcArea();
	}
	calcArea() {
		return this.height * this.width;
	}
	
}

eval('var RectangleY = class RectangleY {constructor(height, width) {this.height = height;this.width = width;}get area() {return this.calcArea();}calcArea() {return this.height * this.width;}}');
var rec1 = new RectangleX(10,20);
var rec2 = new RectangleY(10,20);
var t0 = performance.now();
for (var p = 0; p < 1000000; p++){
	var t = rec1.area;
}
var t1 = performance.now();
console.log("Took " + ((t1-t0)/1000) + " secs");

var t2 = performance.now();
for (var p = 0; p < 1000000; p++){
	var s = rec2.area;
}
var t3 = performance.now();
console.log("Took " + ((t3-t2)/1000) + " secs");
```

Output: 
`Took 5.600000004051253 msecs`
`Took 4.999999975552782 msecs`

**Comment by PedroMiguelSS** [comment]
Regarding my last comment and its possible solution, I've been working on performance testing to realize if _eval()_ function is a feasible solution. 
I did 3 kinds of tests: 

- Class Declaration (with _eval()_ and without _eval()_)
- Instantiate Object of Class (after class declaration)
- Access Instance's Method

The tests were done using [benchmark.js](https://github.com/bestiejs/benchmark.js/) library and were applied on browsers (Chrome, Safari and Firefox) and also on nodejs.

The next tables are screenshots of those tests.

**Class Declaration**
<img width="394" alt="class declaration" src="https://user-images.githubusercontent.com/11461281/38558943-9e9e0afa-3cc9-11e8-841d-e573ab749f96.png">

**Instantiate Object of Class**
<img width="449" alt="instantiate object class" src="https://user-images.githubusercontent.com/11461281/38559144-233687ce-3cca-11e8-9242-20a2b6b1d861.png">

**Access Instance's Method**
<img width="450" alt="instance usage" src="https://user-images.githubusercontent.com/11461281/38559317-9c453b56-3cca-11e8-8587-847ee8e19715.png">

Taking these results into account and considering that _eval()_ function is known as a dangerous function, I think the solution that I provided in the last comment shouldn't be considered anymore.
Note that weird result on Google Chrome when we're instantiating the object (± 57,31%). 

This can be considered an issue on browsers such as Google Chrome and Firefox. However, it works well on Safari and nodeJS. 

To sum it up:
You can logging an instance and get your class name, instead of class wrapper name on: 
- Safari
- NodeJS

You can't do the same thing on:
- Firefox
- Google Chrome (this can be possibly fixed in future updates of chrome V8 engine)

**Comment by fsdiogo** [comment]
Thanks for taking the time @PedroMiguelSS.

This issue is not really an issue, it's a nice to have that has been partially solved with the [toStringTag()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag) method.

I also agree that using `eval` is a bit *funky* to say the least, and it worries me that it could drive away people that would use this package.

That being said, I'm going to close this one for now!
